airbyte-cdk/python/airbyte_cdk/sources/declarative/decoders/json_decoder.py

```diff
@@ -9,5 +9,9 @@
 
 
 class JsonDecoder(Decoder):
+    """
+    Decoder strategy that returns the json-encoded content of a response, if any.
+    """
+
     def decode(self, response: requests.Response) -> Mapping[str, Any]:
-        return response.json()
+        return response.json() or {}
```

Comment:
`or {}` because response.json() can return `None`

Reply:
fwiw I believe `.json()` can also return an array -- probably a follow up ticket since this PR is focused on docs

Reply:
Added `Union` to the type hint and created a follow up issue https://github.com/airbytehq/airbyte/issues/15014